### PR TITLE
Mariam/condition evaluator tests

### PR DESCRIPTION
**Pull request by Mariamjamal32**
Summary
- Added test cases for conditionTreeEvaluator.

### CODE REVIEW

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php [hunk 1]
```diff
@@ -63,35 +64,268 @@ protected function getLeafEvaluator($a, $b = null, $c = null) {
...
             $this->conditionTreeEvaluator->evaluate(
-                ['and', $this->conditionA, $this->conditionB], 
+                ['and', $this->conditionA, $this->conditionB],
                 $this->getLeafEvaluator(True, False)
             )
         );
     }
 
-    public function testAndEvaluatorReturnsNullWhenTruesAndNulls()
+    // Test that andEvaluator returns True when all conditions evaluate to True.
```
Comment (oakbani) [inline tests/UtilsTests/ConditionTreeEvaluatorTest.php]: nit. in comments either use capital case True/False or small. In PHP we can use small. 
We should be consistent across the file

Comment (oakbani) [reply tests/UtilsTests/ConditionTreeEvaluatorTest.php]: Similarly for sending true/false in getLeafEvaluator

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php [hunk 2]
```diff
@@ -63,35 +64,268 @@ protected function getLeafEvaluator($a, $b = null, $c = null) {
...
+
+        $this->assertNull(
+            $this->conditionTreeEvaluator->evaluate(
+                ['and', $this->conditionA, $this->conditionB, $this->conditionC],
+                $this->getLeafEvaluator(null, true, true)
+            )
+        );
+    }
+
+    //  Test that andEvaluator returns False when when operands evaluate to falses and null.
```
Comment (oakbani) [inline tests/UtilsTests/ConditionTreeEvaluatorTest.php]: nit. when

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php [hunk 3]
```diff
@@ -63,35 +64,268 @@ protected function getLeafEvaluator($a, $b = null, $c = null) {
...
+        $this->assertNull(
+            $this->conditionTreeEvaluator->evaluate(
+                ['not'],
+                $this->getLeafEvaluator(null)
+            )
+        );
+    }
+
+    // Test that by default OR operator is assumed when the first item in conditions is not
+    //    a recognized operator.
```
Comment (oakbani) [inline tests/UtilsTests/ConditionTreeEvaluatorTest.php]: nit. spacing

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php (hunk 1 above)
Comment (Mariamjamal32) [reply tests/UtilsTests/ConditionTreeEvaluatorTest.php]: Done.

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php (hunk 3 above)
Comment (Mariamjamal32) [reply tests/UtilsTests/ConditionTreeEvaluatorTest.php]: Removed unnecessary spaces.

File: tests/UtilsTests/ConditionTreeEvaluatorTest.php (hunk 2 above)
Comment (Mariamjamal32) [reply tests/UtilsTests/ConditionTreeEvaluatorTest.php]: Catered.

**Pull request merged**